modules/library/main/src/main/java/org/geotools/feature/AttributeBuilder.java

```diff
@@ -214,15 +215,15 @@ public void setCRS(CoordinateReferenceSystem crs) {
 
     /** @return The coordinate reference system of the feature, or null if not set. */
     public CoordinateReferenceSystem getCRS(Object geom) {
-        if (crs != null) {
-            return crs;
-        } else if (geom != null && geom instanceof Geometry) {
-            Object userData = ((Geometry) geom).getUserData();
-            if (userData != null && userData instanceof CoordinateReferenceSystem) {
-                return (CoordinateReferenceSystem) userData;
+        if (geom != null && geom instanceof Geometry) {
+            // the geometry in the geometry itself is preferred
+            // to support multiple geometries/CRS
+            CoordinateReferenceSystem featureCrs = GeometryUtil.getCRS((Geometry) geom);
+            if (featureCrs != null) {
+                return featureCrs;
             }
```

Comment:
Note: this is the main change in app-schema. The CRS in the data itself will be used if it is available.

Reply:
Does not seem like a big change in behavior, the old code was already picking the CRS from the Geometry value... The changed code introduces a variation, will pick the CRS also from a map in the userdata. Does not seem problematic.

Reply:
The big change in behaviour is the order of preference (see above) which prevents a single CRS to be used for each geometry.